### DISCUSSION TITLE
[Question]: Set up RAG API with npm

### DISCUSSION THREAD

**Discussion by mariannebrissette3** [danny-avila/LibreChat#3744] [Q&A]
### What is your question?

Docker is unsuited for development. I am trying to set up the app to work with npm. The difficulty is to set up the RAG API to work and call it from the app. There is very limited documentation on how to do this here: https://www.librechat.ai/docs/configuration/rag_api#local-setup 
Currently, I am able to run the rag api locally and to call it from the .env file, but the files take forever to load.

### More Details

Please advise on how to work around this issue and use npm to run the app, pursue development, and use files with the RAG API.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
> Currently, I am able to run the rag api locally and to call it from the .env file, but the files take forever to load.

It doesn't take forever on my end, could be a local hardware limitation. 

> Docker is unsuited for development

Right, but you can develop in a python venv without using docker. It's briefly mentioned in the separate repo readme:

  - Local:
    - Make sure to setup `DB_HOST` to the correct database hostname
    - Run the following commands (preferably in a [virtual environment](https://realpython.com/python-virtual-environments-a-primer/))
```bash
pip install -r requirements.txt
uvicorn main:app
```

source: https://github.com/danny-avila/rag_api

**Reply by mariannebrissette3** [reply]
Thanks for your reply! I was able to figure out what my issue is and I can explain it better. The RAG API is correctly installed (I think) and it runs on my machine. I can upload a file in the app and it is stored correctly in the database. However, the retrieval is what does not work. On the RAG logs, I see:
root - ERROR - Error fetching original document: 404: File not found
and the LLM does not see the file content. Any advice on how to proceed then?
